tree/treeplayer/src/TTreeReaderArray.cxx

```diff
@@ -532,7 +532,25 @@ void ROOT::Internal::TTreeReaderArrayBase::CreateProxy()
          return;
       }
 
-      if (fDict != branchActualType) {
+      auto matchingDataType = [](TDictionary *left, TDictionary *right) -> bool {
+         if (left != right) {
```

Comment:
Can you invert? Can you check for nullptr first, or should nullptr==nullptr actually return happiness? If so, could you add a comment on that?


```suggestion
         if (!left || !right)
            return false;
         if (left == right)
            return true;
```

This keeps indentation lower (and for me it also makes the code simpler but that's a matter of taste).

Reply:
Sure.  Thanks.